src/sql/workbench/contrib/notebook/test/testCommon.ts

```diff
@@ -14,24 +16,39 @@ import { TestThemeService } from 'vs/platform/theme/test/common/testThemeService
 import { TestEditorGroupsService, TestEditorService, TestTextResourceConfigurationService } from 'vs/workbench/test/browser/workbenchTestServices';
 import { TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
 
+/**
+ *	Helper method to build get a spied object one can use to listen on using setups, returns and callback and the returned mock object
+ *
+ * @param object - the object to be spied upon
+ * @param behavior - the TypeMoq.MockBehavior value. Default is Loose
+ * @param callBase - whether to forward all calls by default to the base object. Default is true.
+ */
+export function getMockAndSpy<U>(object: U, behavior: TypeMoq.MockBehavior = TypeMoq.MockBehavior.Loose, callBase: boolean = true) {
```

Comment:
return type for this

But I also don't see the value in this function. This isn't doing anything helpful in my eyes - there's no logic here since the user already passes in the required parameters and creating a mock is a 1 or 2 line thing. Nothing we should care about trying to consolidate IMO. 

And actually - do you even use this anywhere? I don't see it being used in this PR at all...

Reply:
it is used in several places. I find this a useful shortcut as I find myself doing this often. Would it be better to put this in my own test class?

Reply:
Well unless I'm mistaken you're not using this in this PR at all. So it should be removed unless you actually use it. 

And I don't see what benefits this brings vs just creating the mocks inline. Most of these options we leave as the default anyways and you can access the object itself directly from the mock so this isn't really saving much that I can see. 

(I also would suggest you don't use spiedObject as the name - spies are a separate thing than mocks)

Reply:
Removed this method